python/paddle/distributed/auto_parallel/static/reshard_funcs/s_to_r_reshard_func.py

```diff
@@ -39,6 +39,35 @@ def is_suitable(self, src_dist_attr, dst_dist_attr):
             return False
         return True
 
+    def infer_allgather_dist_type(self, in_value, split_axis):
+        tensor_ndim = len(in_value.shape)
+        in_dist_attr = in_value.dist_attr()
+        split_mesh_dim = in_dist_attr.dims_mapping[split_axis]
+        mesh = in_dist_attr.process_mesh
+
+        # Calculate local shape. In nd_mesh_reshard, multiple tensor axis
+        # may be shard and it will call this 1-D s_to_r function on each
+        # axis. In this case, we should recompute the local and global shape.
+        out_local_shape = list(in_value.shape)
```

Comment:
it might be better to regard the intermediate results within reshard transformation  as local tensor and there is not need to assign dist_attr to it.